src/main/kotlin/org/wa/rceditor/application/model/MainModel.kt

```diff
@@ -1,26 +1,310 @@
+    val rightsProperty = getProperty(MainModel::rights)
+
+    private var subject: String by property()
+    val subjectProperty = getProperty(MainModel::subject)
+
+    private var title: String by property()
+    val titleProperty = getProperty(MainModel::title)
+
+    private var type: String by property()
+    val typeProperty = getProperty(MainModel::type)
+
+    private var version: String by property()
+    val versionProperty = getProperty(MainModel::version)
+
+    private var checkingLevel: String by property()
+    val checkingLevelProperty = getProperty(MainModel::checkingLevel)
+
+    private val contributors = SortedFilteredList<ContributorItem>()
+    private val relations = SortedFilteredList<RelationItem>()
+    private val sources = SortedFilteredList<SourceItem>()
+    private val checkingEntities = SortedFilteredList<CheckingEntityItem>()
+    private val projects = SortedFilteredList<ProjectItem>()
+
+    private var directoryLoaded: Boolean by property()
+    val directoryLoadedProperty = getProperty(MainModel::directoryLoaded)
+
+    lateinit var container: ResourceContainer
+    lateinit var directory: File
+
+    // -------------- Functions --------------- //
+
+    fun contributors(): ObservableList<ContributorItem> {
+        return contributors
+    }
+
+    fun addContributor(text: String) {
+        contributors.add(ContributorItem(text))
+    }
+
+    fun removeContributor(item: ContributorItem) {
+        contributors.remove(item)
+    }
+
+    fun relations(): ObservableList<RelationItem> {
+        return relations
+    }
+
+    fun addRelation(text: String) {
+        relations.add(RelationItem(text))
+    }
+
+    fun removeRelation(item: RelationItem) {
+        relations.remove(item)
+    }
+
+    fun sources(): ObservableList<SourceItem> {
+        return sources
+    }
+
+    fun addSource(identifier: String, language: String, version: String) {
+        sources.add(SourceItem(identifier, language, version))
+    }
+
+    fun removeSource(item: SourceItem) {
+        sources.remove(item)
+    }
+
+    fun checkingEntities(): ObservableList<CheckingEntityItem> {
+        return checkingEntities
+    }
+
+    fun addCheckingEntity(text: String) {
+        checkingEntities.add(CheckingEntityItem(text))
+    }
+
+    fun removeCheckingEntity(item: CheckingEntityItem) {
+        checkingEntities.remove(item)
+    }
+
+    fun projects(): ObservableList<ProjectItem> {
+        return projects
+    }
+
+    fun addProject(title: String, versification: String, identifier: String, sort: Int, path: String, category: String) {
+        projects.add(ProjectItem(title, versification, identifier, sort, path, category))
+    }
+
+    fun removeProject(item: ProjectItem) {
+        projects.remove(item)
+    }
+
+    // ---------- Handlers --------- //
+
     fun onNewDocumentSelected() {
-        println("New document created")
+        chooseDirectory("Create Resource Container").apply {
```

Comment:
This would be a good candidate for a standalone domain layer use case

Reply:
@mbr4477 I've moved it to the usecases in PR#3